pkg/segment/reader/metrics/series/seriesreader.go

```diff
@@ -333,3 +333,45 @@ func (tssr *TimeSeriesSegmentReader) loadTSGFile(fileName string, rbuf []byte) (
 	}
 	return rbuf, nil
 }
+
+func (tssr *TimeSeriesSegmentReader) GetAllMetricNames() (map[string]bool, error) {
+
+	filePath := fmt.Sprintf("%s.mnm", tssr.mKey)
+
+	fd, err := os.OpenFile(filePath, os.O_RDONLY, 0644)
+	if err != nil {
+		log.Errorf("GetAllMetricNames: failed to open fileName: %v  Error: %v", filePath, err)
+		return nil, err
+	}
+
+	defer fd.Close()
+
+	finfo, err := fd.Stat()
+	if err != nil {
+		log.Errorf("GetAllMetricNames: error when trying to stat file=%+v. Error=%+v", filePath, err)
+		return nil, err
+	}
+
+	fileSize := finfo.Size()
+	buf := make([]byte, fileSize)
```

Comment:
in next PR (todo) , you should try to use buffer pools for such kinds of memory accesses, it will reduce GC pressures.